### ISSUE TITLE
Watch mode broken in beta9

### ISSUE THREAD

**Issue by harrysarson** [rtfeldman/node-test-runner#294]
augustin82 pointed out in slack that the running tests in watch mode fails to actually run the tests.

I believe this is caused by my pull request (sorry!) #279. 

Possibly also caused by @rfeldman's commit 0123787 which removed a promise resolve call that I believe I added as part of my pr and I also believe I added for a reason.

I have not had time to properly look into the error but above is my hunch.

**Comment by harrysarson** [comment]
This can (apparently) be reproduced by running any test suit in watch mode and editing a test file.

**Comment by Augustin82** [comment]
Yes. Sorry, it's hard to be more specific, but what I've done is follow the instructions for elm-explorations/test, tried editing the example test while in watch mode, and it would correctly detect the file was updated, but tests would never rerun.
I downgraded to the beta8 version after seeing here and there it had fixed some problems for others, and watch mode worked as expected from then on.

**Comment by harrysarson** [comment]
@Augustin82 I cannot reproduce this on linux, which os are you using?

**Comment by Augustin82** [comment]
@harrysarson I'm using Ubuntu 18.04. I'll try to reproduce it right now and will report back.

**Comment by Augustin82** [comment]
OK, so, you're right: I managed to have it work with the latest beta. However, I know why I mistakenly reported that error: it's because, on some occurrences, starting `elm-test --watch` does not give the test output, for some reason. So maybe tests were actually re-running, but it didn't show in my console. I took two screenshots to illustrate the difference. The first one worked fine, the second one is the bug.
![selection_067](https://user-images.githubusercontent.com/2370810/46257647-8b50ee00-c4bd-11e8-9f90-05f2be520358.jpg)
![selection_068](https://user-images.githubusercontent.com/2370810/46257651-9ad03700-c4bd-11e8-816a-d848950acb5e.jpg)
